hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java

```diff
@@ -83,6 +84,15 @@ protected HiveMetaStoreClient reconnect(HiveMetaStoreClient client) {
     return client;
   }
 
+  @Override
+  protected boolean failureDetection(Exception e) {
+    if (Objects.nonNull(e) && e instanceof MetaException &&
```

Comment:
nit: Do we need to use `Objects.nonNull(e)`? Maybe I am old fashioned but should `e != null` suffice?

Thanks,
Peter

Reply:
I would replace it with `e != null`